packages/ld_default.rb

```diff
@@ -0,0 +1,61 @@
+require 'package'
+
+class Ld_default < Package
+  description 'Select the default ld executable and check for libraries in #{CREW_LIB_PREFIX} first'
+  homepage 'https://github.com/skycocker/chromebrew'
+  version '1.0'
+  source_url 'https://raw.githubusercontent.com/skycocker/chromebrew/93aa4dc4d9198e28d84dcf9d59b3ab6b6a94d92c/README.md'
+  source_sha256 '3ec9c6f7ba6a138d07dfb4a372d562cd3ad558b58b6bd447c2dbb52e16b697a5'
+
+  depends_on 'llvm'
+
+  if ARGV[0] == 'install'
+    puts
+    Dir.chdir "#{CREW_PREFIX}/bin" do
+      system "touch ld"
+      slashes = "#{CREW_PREFIX}/bin/ld/".count('/')
+      current = `tail -1 ld | cut -d'/' -f#{slashes} 2> /dev/null | cut -d' ' -f1 2> /dev/null`
+      puts "Current default: #{current}"
+    end
+    puts
+    puts "Enter the ld default:"
+    puts "b = ld.bfd"
+    puts "g = ld.gold"
+    puts "l = ld.lld"
+    puts "0 = Cancel"
+
+    while default = STDIN.gets.chomp
+      case default
+      when 'b'
+        $current = "ld.bfd"
+        break
+      when 'g'
+        $current = "ld.gold"
+        break
+      when 'l'
+        $current = "ld.lld"
+        break
+      when '0'
+        abort
+        break
+      else
+        puts "Please select from one of the options or enter 0 to cancel."
+      end
+    end
+    system "echo '#!/bin/bash' > ld"
+    system "echo '#{CREW_PREFIX}/bin/#{$current} -L #{CREW_LIB_PREFIX} -L /usr/#{ARCH_LIB} -L /#{ARCH_LIB} \"$@\"' >> ld"
+    system "install -Dm755 ld #{CREW_PREFIX}/bin/ld"
+  end
+
+  def self.install
+    system "echo '#!/bin/bash' > ld"
+    system "echo '#{CREW_PREFIX}/bin/#{$current} -L #{CREW_LIB_PREFIX} -L /usr/#{ARCH_LIB} -L /#{ARCH_LIB} \"$@\"' >> ld"
```

Comment:
It's not about redundancy, it's about the ORDER the linker determines which libraries to link against.  Let's say I want to link using `/usr/local/lib/library.so` instead of the system `/lib/library.so`.  The way it works now is the system library will be chosen even though the desired library is `/usr/local/lib/library.so`.  If you don't believe me, try and compile vscode from source and see what happens.  I haven't noticed it much but sometimes `ld.gold` works when `ld.bfd` fails.  This seems to happen more often on i686 than any other architecture from my experience.

Reply:
What? `ld.bfd` is much more stable than `ld.gold`. You really could just disable using any system libraries if you just use the `-rpath` option.

Reply:
If you want to, just change the `--with-default-library-search-path=<blah:blah:blah>` and the `--enable-gold` options in `binutils`' configure options.